### DISCUSSION TITLE
Legacy Rootless Jetsam Configuration

### DISCUSSION THREAD

**Discussion by DetachHead** [UInt2048/CyberKit#10] [General]
can the jetsam instructions be done in a postinstall script when installing from sileo?

**Comment by UInt2048** [comment]
A postinstall script could potentially be offered at some point for rootful devices.

My guess is for rootless devices, if jetsam instructions end up working (nobody has tested my idea yet), that it will have to be reapplied after every reboot. A postinstall script probably would not exist on rootless.

**Comment by DetachHead** [comment]
i tried testing your idea but SantanderMacDirtyCow doesn't seem to have a way to create new keys in plist files, are there any other macdirtycow filesystem editors you know of that i could try?

<details>
<summary>SantanderMacDirtyCow screenshot</summary>

![image](https://github.com/UInt2048/CyberKit/assets/57028336/e019adbb-b165-46ad-ba12-a840cb20b6cb)
</details>


i'm not very familiar with iOS tweak development but could this potentially be accomplished using tweak injection instead?

**Comment by UInt2048** [comment]
@DetachHead What if you make a copy of the file to some writable directory like /var/mobile/Documents and edit that with Filza or something and then use SantanderMacDirtyCow to “replace” the system version with your edited one?

If someone were to develop a tweak that can increase jetsam limits that allows rootless users to use CyberKit, I will be happy to add those instructions to the repo. However, I have no rootless devices, so anything I say is a guess that I cannot test. In addition, development of such a tweak is out of scope for the project.

On my iOS 14 device, editing the PLIST directly requires a reboot. I was not successful with using existing tweaks that I could find.(CyberKit requires that the XPC services have higher memory, but doesn’t care about the browser application.)

**Comment by HackZy01** [comment]
Santander MacDirtyCow doesn’t offer the ability to overwrite system files, you can only do this with a jailbreak or TrollStore, but definitely not MacDirtyCow

**Comment by HackZy01** [comment]
The replace a system file with MDC it needs to be smaller or the same file size as the original file

**Comment by UInt2048** [comment]
@HackZy01 Are you saying you could use MDC to replace “com.apple.WebKit.\*” with a shorter string like “uint2048.cyber.\*” in the file?

If so, can you try doing that and see if Safari crashes when you try using it?

**Comment by benjipeng** [comment]
Hello, @UInt2048 I would like to give you a quick update on trying to config `rootless` JetSam under `iOS 15.1.1, Dopamine, iPhone 13 Pro Max`. It looks like Jetsam config may _not_ work for rootless.

**What I Have Done:**
1. Followed your instructions and made a `copy` of `com.apple.jetsamproperties.{xxx}.plist` from `/System/Library/LaunchDaemons`and added JetSam for `com.matthewbenedict.CyberKit.{xxx}`.
2. Used a tool called `mount_bindfs` to bind the `/System/Library/LaunchDaemons` to the `jb` folder.
3. Replaced the updated `jetsamproperties` file in the Binded directory.
4. Restarted device and re-jailbreak.
5. Confirmed the changes persisted in the folder bound under `jb-{xxx}/procursus/System/Library/LaunchDaemons`.

**Result:**
- `CyberKit Fennec iOS 15 Rootless` (from your repo) crashes upon opening.
- `CyberKit MobileMiniBrower iOS 15 Rootless` (from your repo) opens but did _NOT_ load any pages.

**Other Infomation:**
`mount_bindfs` _should_ work because I have also bound the `System/Library/Fonts` and was able to replace my `lock screen fonts`.

**Comment by UInt2048** [comment]
Future releases of CyberKit will perform their own jetsam configuration using syscalls instead of editing a system property list (which was always a bad method anyways).

Consequently, this discussion is resolved as follows:

* CyberKit releases before 0.0.9 (all released versions up to this point) officially do not support rootless jailbreaks.
* CyberKit releases since 0.0.9 do this from the source of CyberKit itself, and thus any problems with this are bugs.

**Reply by ghh-jb** [reply]
No support for iOS 15?
